cpp/daal/src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -129,23 +152,58 @@ services::Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::compute(c
 
     auto dataSumOfSquares = context.allocate(TypeIds::id<algorithmFpType>(), maxDataBlockRowCount, st);
     DAAL_CHECK_STATUS_VAR(st);
+    UniversalBuffer querySumOfSquares;
+    if(computeOutputDistances)
+    {
+        querySumOfSquares = context.allocate(TypeIds::id<algorithmFpType>(), maxQueryBlockRowCount, st);
+        DAAL_CHECK_STATUS_VAR(st);
+    }
     auto distances = context.allocate(TypeIds::id<algorithmFpType>(), maxDataBlockRowCount * maxQueryBlockRowCount, st);
     DAAL_CHECK_STATUS_VAR(st);
     auto partialDistances = context.allocate(TypeIds::id<algorithmFpType>(), maxQueryBlockRowCount * k * selectionMaxNumberOfChunks, st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto partialLabels = context.allocate(TypeIds::id<int>(), maxQueryBlockRowCount * k * selectionMaxNumberOfChunks, st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto sortedLabels = context.allocate(TypeIds::id<int>(), maxQueryBlockRowCount * k, st);
-    DAAL_CHECK_STATUS_VAR(st);
+
+    UniversalBuffer partialLabels;
+    if(computeOutputLabels)
+    {
+        partialLabels = context.allocate(TypeIds::id<int>(), maxQueryBlockRowCount * k * selectionMaxNumberOfChunks, st);
+        DAAL_CHECK_STATUS_VAR(st);
+    }
+    // temporary buffer for indices
+    UniversalBuffer blockIndices;
+    if(computeOutputIndices || isOwnDistances)
+    {
+        blockIndices = context.allocate(TypeIds::id<int>(), maxDataBlockRowCount, st);
+        DAAL_CHECK_STATUS_VAR(st);
+    }
+    UniversalBuffer partialIndices; 
+    if(computeOutputIndices || isOwnDistances)
```

Comment:
You could merge some if(...) blocks

Reply:
It cost almost nothing and we can ignore it to maintain clarity